File: 344-ReverseString/344-ReverseString.py
# Last updated: 2/28/2026, 4:27:32 PM
class Solution:
    def reverseString(self, s: List[str]) -> None:
        """
        Do not return anything, modify s in-place instead.
        """

        l, r = 0, len(s) -1

        while l < r:
            temp = s[l]
            s[l] = s[r]
            s[r] = temp

            l += 1
            r -= 1
        

File: 344-ReverseString/344-ReverseString.java
// Last updated: 2/28/2026, 4:27:34 PM
class Solution {
    public void reverseString(char[] s) {

        solve(s,0,s.length - 1);
        
    }

    public void solve(char[] s, int i , int j){
        if(i >= j) return;

        char temp = s[i];
        s[i] = s[j];
        s[j] = temp;

        solve(s,++i,--j); 
    }
}